frontend/src/metabase/visualizations/components/settings/ChartSettingInputSuggestion.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useRef } from "react";
+import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+import SelectList from "metabase/components/SelectList";
+import { useListKeyboardNavigation } from "metabase/hooks/use-list-keyboard-navigation";
+import {
+  SuggestionInput,
+  SuggestionContainer,
+} from "./ChartSettingInputSuggestion.styled";
+
+interface ChartSettingInputSuggestionProps {
+  value: string;
+  onChange: (value: string) => void;
+  id?: string;
+  options?: string[];
+}
+
+const matchPattern = /.*{{([^{}]*)$/;
```

Comment:
The regex approach works nicely for one of the main cases which will definitely bring a lot of convenience for users. However, in order to support suggestions in the middle of the text, we will need to use `ace` in the future. Anyway, I think it is a good UX improvement given that was suggested and implemented ad-hoc

Reply:
nit: rename this constant to make it more clear what we are matching for. E.g. `linkVariablePattern`